technic_chests/inventory.lua

```diff
@@ -67,6 +69,37 @@ function technic.chests.sort_inv(inv, mode)
 			end
 			return a.key < b.key
 		end)
+	elseif mode == 4 then
+		-- Natural
+		do -- Collect item stacks for sorting
+			local function name(stack)
+				return stack:get_meta():get("infotext")
+					or stack:get_description()
+					or stack:get_name()
+			end
+			local lookup = {}
+			for _,stack in pairs(list) do
+				local key = minetest.get_translated_string('', name(stack))
```

Comment:
Note that here if we had player data available sorting could use translated descriptions, empty locale seems to return default translation which is usually english.
Currently there's no player available when this function is called so players using other locale still gets items sorted by english descriptions.